### PR TITLE
Uplift arm64 universal build (uplift to 1.18.x)

### PR DESCRIPTION
**Pull request by bridiver**
Uplift of #7366, #7475, #7482
Fixes https://github.com/brave/brave-browser/issues/13074

Approved, please ensure that before merging: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 
- [x] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Review by mihaiplesa — APPROVED** [review]

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.18.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
